tests/models/wav2vec2_bert/test_modeling_wav2vec2_bert.py

```diff
@@ -0,0 +1,955 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Testing suite for the PyTorch Wav2Vec2-BERT model. """
+import tempfile
+import unittest
+
+from datasets import load_dataset
+
+from transformers import Wav2Vec2BERTConfig, is_torch_available
+from transformers.testing_utils import (
+    is_pt_flax_cross_test,
+    require_torch,
+    require_torch_accelerator,
+    require_torch_fp16,
+    slow,
+    torch_device,
+)
+
+from ...test_configuration_common import ConfigTester
+from ...test_modeling_common import (
+    ModelTesterMixin,
+    _config_zero_init,
+    floats_tensor,
+    ids_tensor,
+    random_attention_mask,
+)
+from ...test_pipeline_mixin import PipelineTesterMixin
+
+
+if is_torch_available():
+    import torch
+
+    from transformers import (
+        AutoFeatureExtractor,
+        Wav2Vec2BERTForAudioFrameClassification,
+        Wav2Vec2BERTForCTC,
+        Wav2Vec2BERTForPreTraining,
+        Wav2Vec2BERTForSequenceClassification,
+        Wav2Vec2BERTForXVector,
+        Wav2Vec2BERTModel,
+    )
+    from transformers.models.wav2vec2_bert.modeling_wav2vec2_bert import (
+        Wav2Vec2BERTGumbelVectorQuantizer,
+        _compute_mask_indices,
+        _sample_negative_indices,
+    )
+
+
+class Wav2Vec2BERTModelTester:
```

Comment:
Can we add `# Copied from` statements from the Wav2Vec2 Conformer tests? Currently it's a bit difficult to pick-out which of the tests implement new logic